### ISSUE TITLE
when I use fp16 mixed train,the loss is fixed

### ISSUE THREAD

**Issue by yulinhuyang** [KevinMusgrave/pytorch-metric-learning#228]
like this：
distance = distances.CosineSimilarity()
reducer = reducers.ThresholdReducer(low = 0)
loss_func = losses.TripletMarginLoss(margin = 0.2, distance = distance, reducer = reducer)
mining_func = miners.TripletMarginMiner(margin = 0.2, distance = distance, type_of_triplets = "semihard")

![image](https://user-images.githubusercontent.com/13074332/98645210-9baa4700-236c-11eb-9d1d-e1eb225d6df3.png)

but when I try another tirplet loss，which come from reid github ,it works ok，is any diffence with fp16 mixed train？
https://github.com/JDAI-CV/fast-reid/tree/master/fastreid/modeling/losses

**Comment by KevinMusgrave** [comment]
Do you know if it trains properly with fp32?

Also, to simplify things, can you try it without the distance, reducer, and miner? Like this:

```
loss_func = losses.TripletMarginLoss()
```

**Comment by yulinhuyang** [comment]
fp32 is very Ok.I will try  without the distance, reducer, and miner

**Comment by KevinMusgrave** [comment]
Closing for now. I'll reopen if there is clearly a bug.
